systemd/watch.go

```diff
@@ -0,0 +1,114 @@
+package systemd
+
+import (
+	"context"
+	"sync"
+
+	"github.com/virtual-kubelet/node-cli/provider"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+// If a provider implements the Updater interface we will send pings when configmaps and secrets are updated.
+type Updater interface {
+	UpdateConfigMap(context.Context, *corev1.Pod, *corev1.ConfigMap) error
+	UpdateSecret(context.Context, *corev1.Pod, *corev1.Secret) error
+}
+
+type Watcher struct {
+	mu        sync.RWMutex
+	clientset *kubernetes.Clientset
+	// maps are indexed by "namespace.name"
+	configs map[string][]*corev1.Pod
+	secrets map[string][]*corev1.Pod
+}
+
+func newWatcher(clientset *kubernetes.Clientset) *Watcher {
+	return &Watcher{
+		configs:   make(map[string][]*corev1.Pod),
+		secrets:   make(map[string][]*corev1.Pod),
+		clientset: clientset,
+	}
+}
+
+func (w *Watcher) Run(p provider.Provider) error {
+	cmwatcher, err := w.clientset.CoreV1().ConfigMaps(corev1.NamespaceAll).Watch(context.TODO(), metav1.ListOptions{})
```

Comment:
Using the low-level Watch API is dangerous. What happens if a watch is closed because of some network hiccup? Now you have to implement your own recovery logic. While I think this is fine for prototyping, my hopes is that one day [`controller-runtime`](https://github.com/kubernetes-sigs/controller-runtime is used) - yep, in VK too.

Reply:
yes, not the watch is re-established on err....

Reply:
Where?

Reply:
Can you please point me to where the watch is re-established?

Reply:
```go
case event, ok := <-swatcher.ResultChan():
	if !ok {
		swatcher, _ = w.clientset.CoreV1().Secrets(corev1.NamespaceAll).Watch(context.TODO(), metav1.ListOptions{})
		break
}
```

Reply:
this?